sub-packages/bionemo-evo2/src/bionemo/evo2/run/train.py

```diff
@@ -879,7 +1176,14 @@ def train(args: argparse.Namespace) -> nl.Trainer:
         constant_steps=args.constant_steps,
     )
     # This is where the no weight decay condition is applied to the optimizer state.
-    opt = MegatronOptimizerModule(opt_config, sched, no_weight_decay_cond=model_config.hyena_no_weight_decay_cond_fn)
+    if (model_type == "hyena") | (model_type == "mamba"):
+        opt = MegatronOptimizerModule(
+            opt_config,
+            sched,
+            no_weight_decay_cond=model_config.hyena_no_weight_decay_cond_fn,
+        )
+    else:
+        opt = MegatronOptimizerModule(opt_config, sched)
```

Comment:
The consequence of this is you have no way to skip weight decay on your embedding layer. That may or may not matter, but it's interesting that you are doing weight decay on embeddings.